modules/postgresql/main.tf

```
@@ -75,6 +87,15 @@ resource "google_sql_database_instance" "default" {
         }
       }
     }
+    dynamic "insights_config" {
+      for_each = local.insights_config[var.insights_config != null ? "enabled" : "disabled"]
```

This logic seems a little confusing. Can't we do it like this?

```suggestion
      for_each = var.insights_config != null ? [var.insights_config] : []
```